test/unit/com/unifina/signalpath/SignalPathSpec.groovy

```diff
@@ -54,4 +57,32 @@ class SignalPathSpec extends BeanMockingSpecification {
 		signalPath.getRootSignalPath() == top
 	}
 
+	def "cyclic dependency detection"() {
+		Canvas rootCanvas = new Canvas()
+		rootCanvas.id = "root"
+		rootCanvas.save(validate: false)
+
+		SignalPath rootSignalPath = new SignalPath(true)
+		rootSignalPath.init()
+		rootSignalPath.configure([canvasId: rootCanvas.id])
+
+		SignalPath subSignalPath = new SignalPath(false)
+		subSignalPath.init()
+		subSignalPath.setParentSignalPath(rootSignalPath)
+
+		when:
+		subSignalPath.configure([
+			hash: 123,
+			params: [
+		    	[name: "canvas", value: "root"]
+			]
+		])
+
+		then:
+		ModuleException ex = thrown()
+
+		ex.getModuleExceptions()[0] instanceof CyclicCanvasModuleExceptionMessage
+		ex.getModuleExceptions()[0].getModuleId() == 123
+	}
```

Comment:
Since the code should work for arbitrary levels of nesting maybe should add a test for when recursion is more than one level deep?

Reply:
@timoxley @hpihkala I tried writing a test like that, but it fails when calling configure() on the first submodule.

Reply:
@kare does that mean there's a bug?